exp/controllers/awsmachinepool_controller.go

```diff
@@ -421,7 +432,7 @@ func (r *AWSMachinePoolReconciler) reconcileLaunchTemplate(machinePoolScope *sco
 	// Check if the instance tags were changed. If they were, create a new LaunchTemplate.
 	tagsChanged, _, _, _ := tagsChanged(annotation, machinePoolScope.AdditionalTags()) // nolint:dogsled
 
-	needsUpdate, err := ec2svc.LaunchTemplateNeedsUpdate(machinePoolScope, &machinePoolScope.AWSMachinePool.Spec.AWSLaunchTemplate, launchTemplate)
+	needsUpdate, err := ec2svc.LaunchTemplateNeedsUpdate(machinePoolScope, &machinePoolScope.AWSMachinePool.Spec.AWSLaunchTemplate, launchTemplate, userDataHash, launchTemplateUserDataHash)
```

Comment:
```suggestion
	needsUpdate, err := ec2svc.LaunchTemplateNeedsUpdate(machinePoolScope, &machinePoolScope.AWSMachinePool.Spec.AWSLaunchTemplate, launchTemplate)
```

I think instead of sending these, we can handle it here like `tagsChanged` case. Because we are not doing anything special other than comparing if they are the same.

Reply:
This check needs to be done here anyway to decide if `StartASGInstanceRefresh` will be called or not.

Reply:
Agreed to both comments